sql/src/main/java/io/crate/operation/collect/sources/NodeStatsCollectSource.java

```diff
@@ -115,4 +104,29 @@ public boolean apply(DiscoveryNode input) {
             )
         );
     }
+
+    @Nullable
+    private List<String> nodeIds(WhereClause whereClause) {
```

Comment:
since the List is only used for `contains` it would probably be better to use a Set.
List contains has linear performance